corehq/apps/users/bulkupload.py

```diff
@@ -438,8 +438,10 @@ def is_password(password):
                         error = custom_data_validator(data)
                         if error:
                             raise UserUploadError(error)
+                        data = {k: unicode(v) for k, v in data.iteritems()}
                         user.user_data.update(data)
                     if uncategorized_data:
+                        uncategorized_data = {k: unicode(v) for k, v in uncategorized_data.iteritems()}
```

Comment:
hehe, well `str(v)` will also blow up if `v` is a unicode string that contains non-ASCII characters, and I'm assuming that's probably more likely than the opposite. I think what you've done here is correct.

Generally always use unicode unless you are dealing with raw binary data (image file content, for example). If a byte string leaks in somewhere and starts causing errors then we need to track it down to the place where we know the encoding and convert it to unicode there.